### PR TITLE
Ldanzinger/readonly

### PR DESCRIPTION
**Pull request by ldanzinger**
@DonMoJa please review. QML has the readonly keyword for properties. We should consistently use these for properties that we know will be constant.

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Layers/RasterLayerFile/RasterLayerFile.qml [hunk 1]
```diff
@@ -23,10 +23,9 @@ import Esri.ArcGISExtras 1.1
 RasterLayerFileSample {
     id: rootRectangle
     clip: true
-
     
     property string dataPath: System.userHomePath + "/ArcGIS/Runtime/Data/raster/"
```
Comment (DonMoJa) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/RasterLayerFile/RasterLayerFile.qml]: readonly missing

File: ArcGISRuntimeSDKQt_CppSamples/Layers/RasterRgbRenderer/RasterRgbRenderer.qml [hunk 2]
```diff
@@ -24,15 +24,14 @@ import Esri.ArcGISExtras 1.1
 RasterRgbRendererSample {
     id: rootRectangle
     clip: true
-
     width: 800
     height: 600
     
     property string dataPath: System.userHomePath + "/ArcGIS/Runtime/Data/raster"
```
Comment (DonMoJa) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/RasterRgbRenderer/RasterRgbRenderer.qml]: same as above

File: ArcGISRuntimeSDKQt_CppSamples/Layers/RasterStretchRenderer/RasterStretchRenderer.qml [hunk 3]
```diff
@@ -23,16 +23,14 @@ import Esri.ArcGISExtras 1.1
 RasterStretchRendererSample {
     id: rootRectangle
     clip: true
-
     width: 800
     height: 600
-
     
     property string dataPath: System.userHomePath + "/ArcGIS/Runtime/Data/raster"
```
Comment (DonMoJa) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/RasterStretchRenderer/RasterStretchRenderer.qml]: same

File: ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceRaster/DynamicWorkspaceRaster.qml [hunk 4]
```diff
@@ -26,7 +26,7 @@ DynamicWorkspaceRasterSample {
     clip: true
 
     property string dataPath: System.userHomePath + "/ArcGIS/Runtime/Data/raster"
```
Comment (DonMoJa) [inline ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceRaster/DynamicWorkspaceRaster.qml]: same

File: ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceShapefile/DynamicWorkspaceShapefile.qml [hunk 5]
```diff
@@ -24,10 +24,9 @@ import Esri.ArcGISExtras 1.1
 DynamicWorkspaceShapefileSample {
     id: rootRectangle
     clip: true
-
     
     property string dataPath: System.userHomePath + "/ArcGIS/Runtime/Data/shapefile"
```
Comment (DonMoJa) [inline ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceShapefile/DynamicWorkspaceShapefile.qml]: same

File: ArcGISRuntimeSDKQt_CppSamples/Search/SearchDictionarySymbolStyle/SearchDictionarySymbolStyle.qml [hunk 6]
```diff
@@ -26,11 +26,10 @@ SearchDictionarySymbolStyleSample {
...
     
-    property double fontSize: 16
-    property var repeaterModel: ["Names", "Tags", "Symbol Classes", "Categories", "Keys"]
-    property var hintsModel: ["Fire", "Sustainment Points", "3", "Control Measure", "25212300_6"]
+    readonly property double fontSize: 16
+    readonly property var repeaterModel: ["Names", "Tags", "Symbol Classes", "Categories", "Keys"]
+    readonly property var hintsModel: ["Fire", "Sustainment Points", "3", "Control Measure", "25212300_6"]
     property var searchParamList: [[],[],[],[],[]]
 
     property url dataPath: System.userHomePath + "/ArcGIS/Runtime/Data/styles/mil2525d.stylx"
```
Comment (DonMoJa) [inline ArcGISRuntimeSDKQt_CppSamples/Search/SearchDictionarySymbolStyle/SearchDictionarySymbolStyle.qml]: same

**Comment by ldanzinger** [comment]
@DonMoJa I'm leaving these as-is, because I have a new PR I'm working on now, where we remove those properties all together from the C++ samples. The goal is to remove the ArcGIS.Extras dependency from C++ samples to make them more portable.

**Pull request merged**